app/src/main/scala/org/alephium/app/ServerUtils.scala

```diff
@@ -334,6 +334,16 @@ class ServerUtils(implicit
         .map(failedInIO)
     } yield BlockEntry.from(block, height)
 
+  def isBlockInMainChain(blockFlow: BlockFlow, blockHash: BlockHash): Boolean = {
+    val result = for {
+      height <- blockFlow.getHeight(blockHash)
```

Comment:
should we return a `NotFound` if the hash doesn't exist?

Reply:
Now it will return an error to be consistent with other endpoints.